cmd/utils/cmd.go

```diff
@@ -332,8 +335,13 @@ func ExportHistory(bc *core.BlockChain, dir string, first, last, step uint64) er
 		if err := w.Finalize(); err != nil {
 			return fmt.Errorf("export failed to finalize %s: %w", fn, err)
 		}
-		log.Info("Exported blockchain to", "dir", dir, "fn", fn)
+		if time.Since(reported) >= 10*time.Second {
+			log.Info("Exporting blocks", "exported", i, "elapsed", common.PrettyDuration(time.Since(start)))
+			reported = time.Now()
+		}
+		fh.Close()
```

Comment:
Not sufficient, you have other `return`s. So either you can salt with `fh.Close()` at all return locations, you you can turn the whole clause within the outer `for` loop into a function -- and then you can use `defer` again.

Reply:
ah yes duh - okay I put it into a function now :+1: